pyproject.toml

```diff
@@ -1,3 +1,99 @@
+[tool.poetry]
+name = "baseplate"
+version = "2.7.0a1"
+description = "reddit's python service framework"
+authors = ["reddit"]
+license = "BSD"
+readme = "README.md"
+homepage = "https://github.com/reddit/baseplate.py"
+repository = "https://github.com/reddit/baseplate.py"
+documentation = "https://baseplate.readthedocs.io/en/stable/"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: POSIX :: Linux",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Software Development :: Libraries :: Application Frameworks",
+]
+
+[tool.poetry.dependencies]
+python = ">=3.8.1"
+advocate = { version = ">=1.0.0,<2.0", optional = true }
+boto3 = ">=1.28.27"
+cassandra-driver = { version = ">=3.29.0,<4.0", optional = true }
+confluent-kafka = { version = ">=2.3.0", optional = true }
+gevent = ">=23.9.1"
+kazoo = { version = ">=2.5.0,<3.0", optional = true }
+kombu = { version = ">=5.3.3", optional = true }
+objgraph = { version = ">=3.6.0", optional = true }
+posix-ipc = ">=1.0.0,<2.0"
+prometheus-client = ">=0.12.0"
+pymemcache = { version = ">=1.3.0,<1.4.4", optional = true }
+pyramid = { version = ">=1.10.8,<2.0", optional = true }
+python-json-logger = ">=2.0.5,<3.0"
+reddit-cqlmapper = { version = ">=0.3.0,<1.0", optional = true }
+redis = { version = ">=2.10.0,<4.0.0", optional = true }
+redis-py-cluster = { version = ">=2.1.2,<3.0.0", optional = true }
+requests = ">=2.21.0,<3.0"
+sentry-sdk = { version = ">=1.35.0,<2.0", optional = true }
+sqlalchemy = { version = ">=1.4.49,<2", optional = true }
+thrift-unofficial = ">=0.19.0,<1.0"
+
+[tool.poetry.extras]
+amqp = ["kombu"]
+cassandra = ["cassandra-driver"]
+cqlmapper = ["reddit-cqlmapper"]
+kafka = ["confluent-kafka"]
+memcache = ["pymemcache"]
+prometheus = []  # Kept for backwards compatibility, these are now main requirements
+pyramid = ["pyramid"]
+redis = ["redis"]
+redis-py-cluster = ["redis-py-cluster"]
+refcycle = ["objgraph"]
+requests = ["advocate"]
+s3fetcher = []  # Kept for backwards compatibility, these are now main requirements
+sentry = ["sentry-sdk"]
+sql = ["sqlalchemy"]
+zookeeper = ["kazoo"]
+
+[tool.poetry.group.dev.dependencies]
+black = "21.10b0"
+# TODO: This can be removed once we upgrade to a newer Black version.
+# https://github.com/psf/black/issues/2964
+click = "<8.1.0"
+fakeredis = "*"
+flake8 = ">=7.0.0"
+lxml = "*"
+moto = "*"
+mypy = "*"
+pre-commit = "*"
+pydocstyle = "*"
+pylint = "*"
+pytest = "7.4.4"
+pytest-cov = "*"
+pytz = "*"
+reorder-python-imports = "2.4.0"
+sphinx = "*"
+sphinx-autodoc-typehints = "*"
+types-redis = "*"
+types-requests = "*"
+types-setuptools = "*"
+webtest = "*"
```

Comment:
dev dependencies can be pinned real strict with no concerns for downstream effects. i worry a bit because too low a version of stuff like `mypy` and `pylint` will _for sure_ conflict w styles.

also: pytz is a direct dev dep?!?!

Reply:
We could set lower bounds (like `>=1.9.0` for mypy or whatever) if you think that's valuable. The lockfile ensures things are pinned, though in theory future upgrades can cause stuff to get downgraded lower than what we might want if there are conflicts. I'm not sure where we'd want to set the bounds though -- it's not uncommon for older Python versions to require older versions of packages than newer versions can use.